contracts/src/tokens/drop/IDropFacet.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+// interfaces
+
+// libraries
+
+// contracts
+
+interface IDropFacetBase {
+  /// @notice A struct representing a claim
+  /// @param conditionId The ID of the claim condition
+  /// @param account The address of the account that claimed
+  /// @param quantity The quantity of tokens claimed
+  struct Claim {
+    uint256 conditionId;
+    address account;
+    uint256 quantity;
+    bytes32[] proof;
+  }
+
+  /// @notice A struct representing a claim condition
+  /// @param currency The currency to claim in
+  /// @param startTimestamp The timestamp at which the claim condition starts
+  /// @param endTimestamp The timestamp at which the claim condition ends
+  /// @param penaltyBps The penalty in basis points for early withdrawal
+  /// @param maxClaimableSupply The maximum claimable supply for the claim condition
+  /// @param supplyClaimed The supply already claimed for the claim condition
+  /// @param merkleRoot The merkle root for the claim condition
+  struct ClaimCondition {
+    address currency;
+    uint40 startTimestamp;
+    uint40 endTimestamp;
+    uint16 penaltyBps;
+    uint256 maxClaimableSupply;
+    uint256 supplyClaimed;
+    bytes32 merkleRoot;
+  }
+
+  // =============================================================
+  //                           Events
+  // =============================================================
+  event DropFacet_Claimed_WithPenalty(
+    uint256 indexed conditionId,
+    address indexed claimer,
+    address indexed account,
+    uint256 amount
+  );
+
+  event DropFacet_Claimed_And_Staked(
+    uint256 indexed conditionId,
+    address indexed claimer,
+    address indexed account,
+    uint256 depositId
+  );
+
+  event DropFacet_ClaimConditionsUpdated(
```

Comment:
I don't think event names need contract prefix.

Reply:
they do if you don't know which facet the event is coming from in a diamond

Reply:
It shouldn't matter. Events are filtered offchain given their signature. There won't be another facet emitting `ClaimedAndStaked` on the same diamond. If there is, all `ClaimedAndStaked` event logs should be indexed.
Prefix is only useful on custom errors to distinguish between `A_InvalidAddress` and `B_InvalidAddress`.